chore: convert all files in developer/17.0/guides/develop (#1104)
keymanapp/help.keyman.com: developer/17.0/guides/develop/tutorial/step-3.md

---
title: Step 3: The Keyboard Header
---

## Comments

```keyman
c Simplified French Keyboard for Keyman 9.0
```

Most of the header in this example is made up of
[comments](/developer/language/guide/comments){.link}. A comment is used
to make notes about the keyboard, or to provide information on the
workings of the keyboard. The comments are readable by anyone looking at
the source code of the keyboard.

A comment always starts with a lowercase ` c `, followed by one or more
spaces, and continues to the end of the line. Keyman Developer will
ignore comments when compiling a keyboard.

Comments can take up a whole line, or can start in the middle of the
line. The latter is useful for making short notes about individual
lines. As you can see we have used both kinds of comments in the header.

## The `&Version` store

```keyman
store(&Version) "9.0"             c This keyboard is for use with Keyman 9.0
```

The [`&Version` store](/developer/language/reference/version){.link}
identifies the Keyman version for which this keyboard was written; this

> keyboard is for use with Keyman 9.0. The `&Version` store is a required
> part of each keyboard header, and should be the first store in the file.

Reviewer: ```suggestion
keyboard is for use with Keyman 9.0. The `&Version` store is an optional
part of the keyboard header, but if present, it should be the first
store in the file.
```

keyboard is for use with Keyman 9.0. The `&Version` store is an optional
part of the keyboard header, but if present, it should be the first
store in the file.

## The `&Name` store

```keyman
store(&Name)    "Quick French"
```

The [`&Name` store](/developer/language/reference/name) specifies
a descriptive name for the keyboard, which can be up to eighty
characters long. The name we have given to this keyboard is
`"Quick French"`. The `&Name` store is not required but is highly
recommended!

## The `&Bitmap` store

```keyman
store(&Bitmap)  "qfrench.ico"
```

The optional [`&Bitmap` store](/developer/language/reference/bitmap)
tells Keyman which image to use for the keyboard's icon. The picture
should be in the standard Windows .ico format, and should contain at
least a single 16x16 pixel image. It can also contain higher resolution
images for high resolution "High DPI" displays. If you use a modern icon
editor, the icon can use alpha transparency. For this keyboard we will
be using the following bitmap:
![](/cdn/dev/img/developer/100/tutorial_keyboard_qfrench.gif); it is
found in the Keyman Developer folder, under

> `Samples\Examples\qfrench.ico` - you should copy it into the same folder

Author: @mcdurdin, the location specified here (Samples/Examples/qfrench.ico) is no longer available from what I can see. Is this outdated and should be instructed to something else?

Reviewer: Good point. Perhaps open a separate issue on this because we may need to re-add it to the installer

Reviewer 2: > Good point. Perhaps open a separate issue on this because we may need to re-add it to the installer

See keymanapp/keyman#11031
